### PR TITLE
Answer:1 - angular-projection

### PR DESCRIPTION
**Pull request by Clarke96**
Didn't create a custom directive for this. I'm using a generic type for the CardComponent so that different array types can be inputted. I can't seem to get the templateRef content to be strict on type checking from the consumer component (city/student/teacher)

### CODE REVIEW

File: apps/angular/projection/src/app/ui/card/card.component.ts [hunk 1]
```diff
@@ -1,34 +1,32 @@
...
+      }
+    `,
+  ],
 })
-export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
+export class CardComponent<T> {
+  @Input() list: T[] = [];
   @Input() customClass = '';
```
Comment (gremlin896) [inline apps/angular/projection/src/app/ui/card/card.component.ts:28]:
```suggestion
```

File: apps/angular/projection/src/app/ui/card/card.component.html [hunk 2]
```diff
@@ -1,27 +1,17 @@
...
 
   <section>
-    <app-list-item
+    <ng-container
       *ngFor="let item of list"
-      [name]="item.firstname"
-      [id]="item.id"
-      [type]="type">
-    </app-list-item>
+      [ngTemplateOutlet]="listItem"
```
Comment (gremlin896) [inline apps/angular/projection/src/app/ui/card/card.component.html:7]: 😎

File: apps/angular/projection/src/app/ui/card/card.component.ts [hunk 3]
```diff
@@ -1,34 +1,32 @@
...
+} from '@angular/core';
+
 import { ListItemComponent } from '../list-item/list-item.component';
 
 @Component({
   selector: 'app-card',
   templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgIf, NgFor, ListItemComponent, NgTemplateOutlet],
```
Comment (gremlin896) [inline apps/angular/projection/src/app/ui/card/card.component.ts:17]:
```suggestion
  imports: [NgFor, NgTemplateOutlet],
```

**Review by gremlin896 — COMMENTED** [review]
Nice solution!

File: apps/angular/projection/src/app/ui/card/card.component.ts (hunk 1 above)
Comment (Clarke96) [reply apps/angular/projection/src/app/ui/card/card.component.ts:28]: 🤦‍♂️

File: apps/angular/projection/src/app/ui/card/card.component.ts (hunk 3 above)
Comment (Clarke96) [reply apps/angular/projection/src/app/ui/card/card.component.ts:17]: 🤦‍♂️

**Pull request closed without merging**